src/buffer/out/AttrRow.hpp

```diff
@@ -20,36 +20,26 @@ Revision History:
 
 #pragma once
 
+#include "til/rle.h"
+
 #include "TextAttributeRun.hpp"
-#include "AttrRowIterator.hpp"
 
-class ATTR_ROW final
+class ATTR_ROW final : public til::rle<TextAttribute, UINT>
 {
 public:
-    using const_iterator = typename AttrRowIterator;
-
-    ATTR_ROW(const UINT cchRowWidth, const TextAttribute attr)
-    noexcept;
+    using mybase = til::rle<TextAttribute, UINT>;
 
-    ~ATTR_ROW() = default;
+    using const_iterator = mybase::const_iterator;
+    using const_reverse_iterator = mybase::const_reverse_iterator;
 
-    ATTR_ROW(const ATTR_ROW&) = default;
-    ATTR_ROW& operator=(const ATTR_ROW&) = default;
-    ATTR_ROW(ATTR_ROW&&)
-    noexcept = default;
-    ATTR_ROW& operator=(ATTR_ROW&&) noexcept = default;
+    using mybase::mybase; // use base constructor
 
     void Reset(const TextAttribute attr);
 
     TextAttribute GetAttrByColumn(const size_t column) const;
     TextAttribute GetAttrByColumn(const size_t column,
                                   size_t* const pApplies) const;

```

Comment:
Might be a good change to finally say goodbye to "Hungarian notation" names.

Reply:
I believe this is slated to be removed/deleted by @DHowett when he finishes hiding everything inside of ROW per #8446 so I didn't want to move the cheese too far on this specific class.